javalib/src/main/scala/java/lang/Utils.scala

```diff
@@ -192,4 +192,9 @@ private[java] object Utils {
     import js.DynamicImplicits.number2dynamic
     (x >>> 0).asInstanceOf[scala.Double]
   }
+
+  /** Round up to the next highest power of 2, if overflow, returns the given number. */
+  @inline def nextPowerOfTwo(i: Int): Int =
+    if (i > (1 << 30)) i
+    else ((1 << 31) >>> (Integer.numberOfLeadingZeros(i - 1)) - 1)
```

Comment:
1 << (32 - Integer.numberOfLeadingZeros(i)

Reply:
That is not correct. It only returns the same result for exact powers of 2. For other values we must round *up* but this formula rounds *down*.

Reply:
yes, maybe:
if (i <= 0) 1 else if (i > 0x40000000) 0x40000000 else 1 << (32 - Integer.numberOfLeadingZeros(i - 1))

Reply:
Thanks for your suggestion.
Well, the use-case of this function is for ensuring the size of internal buffers (e.g. internal Array of `ArrayList` and `PriorityQueue`).
Therefore,

- this function should always return a number **larger than or equel to** the given number (so we shouldn't have `if (i > 0x40000000) 0x40000000`)
- and for those specific use cases, I think we can assume the input is larger than 0.  With this assumption, we can ignore the case of `i <= 0` and remove a branch.